dts/bindings/adc/adc-comparator.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Comparator will monitor signal though ADC and based on configuration,
+    it will trigger a callback to user to inform signal criterion is met.
```

Comment:
Callbacks are a software feature. Bindings are a hardware description language. Please rephrase this documentation in terms of the hardware, not the higher level software that happens to be using the binding.

Reply:
I will rephrase this.